aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java

```diff
@@ -0,0 +1,211 @@
+    String location = glueCatalog.defaultWarehouseLocation(TableIdentifier.of("db", "table"));
+    Assert.assertEquals("s3://bucket2/db/table", location);
+  }
+
+  @Test
+  public void listTables() {
+    Mockito.doReturn(GetDatabaseResponse.builder()
+        .database(Database.builder().name("db1").build()).build())
+        .when(glue).getDatabase(Mockito.any(GetDatabaseRequest.class));
+    Mockito.doReturn(GetTablesResponse.builder()
+        .tableList(
+            Table.builder().databaseName("db1").name("t1").build(),
+            Table.builder().databaseName("db1").name("t2").build()
+        ).build())
+        .when(glue).getTables(Mockito.any(GetTablesRequest.class));
+    Assert.assertEquals(
+        Lists.newArrayList(
+            TableIdentifier.of("db1", "t1"),
+            TableIdentifier.of("db1", "t2")
+        ),
+        glueCatalog.listTables(Namespace.of("db1"))
+    );
+  }
+
+  @Test
+  public void dropTable() {
+    Mockito.doReturn(GetDatabaseResponse.builder()
+        .database(Database.builder().name("db1").build()).build())
+        .when(glue).getDatabase(Mockito.any(GetDatabaseRequest.class));
+    Mockito.doReturn(DeleteTableResponse.builder().build())
+        .when(glue).deleteTable(Mockito.any(DeleteTableRequest.class));
+    glueCatalog.dropTable(TableIdentifier.of("db1", "t1"));
+  }
+
+  @Test
+  public void renameTable() {
+    Mockito.doReturn(GetDatabaseResponse.builder()
+        .database(Database.builder().name("db1").build()).build())
+        .when(glue).getDatabase(Mockito.any(GetDatabaseRequest.class));
+    Mockito.doReturn(DeleteTableResponse.builder().build())
+        .when(glue).deleteTable(Mockito.any(DeleteTableRequest.class));
+    Mockito.doReturn(CreateTableResponse.builder().build())
+        .when(glue).createTable(Mockito.any(CreateTableRequest.class));
+    glueCatalog.dropTable(TableIdentifier.of("db1", "t1"));
+  }
+
+  @Test
+  public void createNamespace() {
+    Mockito.doReturn(CreateDatabaseResponse.builder().build())
+        .when(glue).createDatabase(Mockito.any(CreateDatabaseRequest.class));
+    glueCatalog.createNamespace(Namespace.of("db"));
+  }
+
+  @Test(expected = NoSuchNamespaceException.class)
+  public void createNamespace_badName() {
+    Mockito.doReturn(CreateDatabaseResponse.builder().build())
+        .when(glue).createDatabase(Mockito.any(CreateDatabaseRequest.class));
+    glueCatalog.createNamespace(Namespace.of("db-1"));
+  }
+
+  @Test
+  public void listNamespaces_all() {
+    Mockito.doReturn(GetDatabasesResponse.builder()
+        .databaseList(
+            Database.builder().name("db1").build(),
+            Database.builder().name("db2").build()
+        ).build())
+        .when(glue).getDatabases(Mockito.any(GetDatabasesRequest.class));
+    Assert.assertEquals(
+        Lists.newArrayList(
+            Namespace.of("db1"),
+            Namespace.of("db2")
+        ),
+        glueCatalog.listNamespaces()
+    );
```

Comment:
I don't really see much value in this test given how heavily mocked it is.

Reply:
100% agree, I am thinking about also publish the actual Glue tests, but I have to also mark them as Ignored because it needs AWS credentials to run.